### ISSUE TITLE
Deprecate in favor of tasty?

### ISSUE THREAD

**Issue by Bodigrim** [haskell/test-framework#62]
This package is on life support for at least five years and requires regular bumps from Hackage trustees to be compatible with the latest ecosystem. It's only a matter of time when revisions would no longer be enough. I'd like to mark it deprecated in favor of `tasty` (since migration from `test-framework` to `tasty` is mostly a matter of changing imports) to give a clear signal to clients that it's time to move on.

**Comment by sol** [comment]
Why should we specifically recommend `tasty` over `hspec` or `HTF` from a repository that is hosted under the `haskell` organization?

**Comment by chessai** [comment]
Right. I don't think we should recommend a single thing, but just deprecate and say "use something else", possibly with a list of alternatives.

**Comment by sol** [comment]
As long as that list is ordered by popularity I'm fine :rofl:

**Comment by Bodigrim** [comment]
@sol as I said, that's because "migration from `test-framework` to `tasty` is mostly a matter of changing imports". `tasty` was built as a successor to `test-framework`, and design decisions are very similar. With all due respect, migrating to `hspec` requires a complete rewrite, but I don't mind mentioning it as an option.

**Comment by sol** [comment]
@Bodigrim not putting unnecessary burden on existing users of `test-framework` makes sense.  Here is my perspective:

- `test-framework` has been stable for years,  I don't see a compelling technical reason for urging people that are happy with `test-framework` to move.
- Looking at the commit history, maintenance overhead seem to be pretty low.  I'm happy to take over maintenance if that helps. **
- Competition is a good thing.  Lack of competition leads to lower quality and hurts the end user (we have a lot of historic evidence for this).

** Conflict of interests: I'm also the maintainer of `hspec`, but I've been maintaining `HUnit` for the last 7 years without biasing it towards `hspec` in any way.
